[CBRD-20523] Test cases for Foreign Key Referential Constraints check on INSERT/UPDATE

kisoo-han opened the pull request:
This is TestCases for [CBRD-20523] Foreign Key Referential Constraints check on INSERT/UDATE.
We changed the rule to check 'Foreign Key Referential Constraints' on INSERT/UPDATE
So, apply of this test case to former CUBRID will lead to verdict 'fail'

eseokoh commented on sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_01.sql:
```diff
@@ -0,0 +1,46 @@
+-- CREATE two TABLEs, Parent/Child Respectively
+-- 
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (
```
To avoid a conflict with other cases, we have kept a rule that `CREATE TABLE` follows `DROP TABLE IF EXIST`.

```sql
DROP TABLE IF EXISTS a_tbl;
CREATE TABLE a_tbl (
  ...
```

eseokoh commented on sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_01.sql:
```diff
@@ -0,0 +1,46 @@
+CONSTRAINT pk_id PRIMARY KEY (id,name_id)
+);
+-- 2. CREATE Child TABLE WITH FOREIGN KEY
+CREATE TABLE b_tbl (
```
Please also add `DROP TABLE IF EXISTS`:

```sql
DROP TABLE IF EXISTS b_tbl;
CREATE TABLE b_tbl (
  ...
```

eseokoh commented on sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_02.sql:
```diff
@@ -0,0 +1,53 @@
+-- So, apply of this test case to former CUBRID will lead to verdict 'fail'
+--
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (
```
Please also add `DROP TABLE IF EXISTS`.

eseokoh commented on sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_02.sql:
```diff
@@ -0,0 +1,53 @@
+CONSTRAINT pk_id PRIMARY KEY (id,name_id)
+);
+-- 2. CREATE Child TABLE WITH FOREIGN KEY
+CREATE TABLE b_tbl (
```
Please also add `DROP TABLE IF EXISTS`.

kisoo-han commented:
add line 'drop table if exist' before the line 'create table'.

kisoo-han commented:
Thank you for your detailed review of code & test cases.
As you requested, I put a line for ‘DROP Table if EXIST’ before ‘CREATE’,
And made a Pull Request on GitHub for curbrid-testcases.

Thank you for your advise again.
--
한기수 (연구소/기술위원)
Tel 070-4367-5140  Fax 070-8650-2150  Mobile 010-5213-2570
Email kisoo-han@cubrid.com<mailto:kisoo-han@cubrid.com>  Facebook http://www.facebook.com/CUBRID.kr
[cid:80B9D229-A73D-495A-ABE8-19095BE68FC2]

From: ESeok Oh <notifications@github.com<mailto:notifications@github.com>>
Reply-To: CUBRID/cubrid-testcases <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: 2016년 9월 30일 금요일 오후 4:34
To: CUBRID/cubrid-testcases <cubrid-testcases@noreply.github.com<mailto:cubrid-testcases@noreply.github.com>>
Cc: HAN Kisoo <kisoo-han@cubrid.com<mailto:kisoo-han@cubrid.com>>, Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [CUBRID/cubrid-testcases] [CBRD-20523] Test cases for Foreign Key Referential Constraints check on INSERT/UPDATE (#36)


@eseokoh requested changes on this pull request.

________________________________

In sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_01.sql<https://github.com/CUBRID/cubrid-testcases/pull/36#pullrequestreview-2280142>:

> @@ -0,0 +1,46 @@
+-- TEST CASE For:
+-- [CBRD-20523] Foreign Key Referential Constraints check on INSERT
+-- Notice:
+-- We changed the rule to check 'Foreign Key Referential Constraints' on INSERT/UPDATE
+-- So, apply of this test case to former CUBRID will lead to verdict 'fail'
+--
+--
+-- CREATE two TABLEs, Parent/Child Respectively
+--
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (


To avoid a conflict with other cases, we have kept a rule that CREATE TABLE follows DROP TABLE IF EXIST.

DROP TABLE IF EXISTS a_tbl;
CREATE TABLE a_tbl (
  ...

________________________________

In sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_01.sql<https://github.com/CUBRID/cubrid-testcases/pull/36#pullrequestreview-2280142>:

> +-- Notice:
+-- We changed the rule to check 'Foreign Key Referential Constraints' on INSERT/UPDATE
+-- So, apply of this test case to former CUBRID will lead to verdict 'fail'
+--
+--
+-- CREATE two TABLEs, Parent/Child Respectively
+--
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (
+ID INT,
+name_id INT,
+phone VARCHAR(10),
+CONSTRAINT pk_id PRIMARY KEY (id,name_id)
+);
+-- 2. CREATE Child TABLE WITH FOREIGN KEY
+CREATE TABLE b_tbl (


Please also add DROP TABLE IF EXISTS:

DROP TABLE IF EXISTS b_tbl;
CREATE TABLE b_tbl (
  ...

________________________________

In sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_02.sql<https://github.com/CUBRID/cubrid-testcases/pull/36#pullrequestreview-2280142>:

> @@ -0,0 +1,53 @@
+-- TEST CASE For:
+-- [CBRD-20523] Foreign Key Referential Constraints check on UPDATE
+-- Notice:
+-- We changed the rule to check 'Foreign Key Referential Constraints' on INSERT/UPDATE
+-- So, apply of this test case to former CUBRID will lead to verdict 'fail'
+--
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (


Please also add DROP TABLE IF EXISTS.

________________________________

In sql/_30_banana_pie_qa/issue_20523_foreign_key/cases/20523_02.sql<https://github.com/CUBRID/cubrid-testcases/pull/36#pullrequestreview-2280142>:

> @@ -0,0 +1,53 @@
+-- TEST CASE For:
+-- [CBRD-20523] Foreign Key Referential Constraints check on UPDATE
+-- Notice:
+-- We changed the rule to check 'Foreign Key Referential Constraints' on INSERT/UPDATE
+-- So, apply of this test case to former CUBRID will lead to verdict 'fail'
+--
+-- 1. CREATE Parent TABLE
+CREATE TABLE a_tbl (
+ID INT,
+name_id INT,
+phone VARCHAR(10),
+CONSTRAINT pk_id PRIMARY KEY (id,name_id)
+);
+-- 2. CREATE Child TABLE WITH FOREIGN KEY
+CREATE TABLE b_tbl (


Please also add DROP TABLE IF EXISTS.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/CUBRID/cubrid-testcases/pull/36#pullrequestreview-2280142>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ASyq1rBpuhbs8fazYk5kH74rRhjNyFEyks5qvLuYgaJpZM4KKxq4>.

eseokoh commented:
It looks good. Please "Squash and merge" the change. You can find the guide to Squash merge from https://help.github.com/articles/merging-a-pull-request/
Thank you. :)
